File: packages/rook_sdk_samsung_health/android/src/main/kotlin/io/tryrook/rook_sdk_samsung_health/mapper/ConfigurationMappers.kt
package io.tryrook.rook_sdk_samsung_health.mapper

import io.tryrook.rook_sdk_samsung_health.extension.checkNonDefault
import io.tryrook.rook_sdk_samsung_health.proto.ConfigurationProto
import io.tryrook.sdk.samsung.domain.model.SHConfiguration
import toSHEnvironment

fun ConfigurationProto.toSHConfiguration(): SHConfiguration {
    return SHConfiguration(
        clientUUID = clientUUID,
        secret = secret,
        environment = environment.toSHEnvironment(),
        packageName = appId.checkNonDefault(),
    )
}

File: packages/rook_sdk_apple_health/lib/src/data/mapper/configuration_mappers.dart
import 'package:rook_sdk_apple_health/src/data/mapper/environment_mappers.dart';
import 'package:rook_sdk_apple_health/src/data/proto/protos.pb.dart';
import 'package:rook_sdk_apple_health/src/domain/util/check_non_default.dart';
import 'package:rook_sdk_core/rook_sdk_core.dart';

extension RookConfigurationMappers on RookConfiguration {
  ConfigurationProto toProto() {
    return ConfigurationProto(
      clientUUID: clientUUID,
      secret: secret,
      environment: environment.toProto(),
      enableBackgroundSync: enableBackgroundSync,
      appId: appId ?? defaultString,
    );
  }
}
